lib/Runtime/PlatformAgnostic/Platform/Linux/DateTime.cpp

```diff
@@ -0,0 +1,195 @@
+            struct tm ltime;
+
+            tzset();
+            localtime_r(&gtime, &ltime);
+
+            zoneNameLength = strlen(ltime.tm_zone);
+
+        #if defined(WCHAR_IS_CHAR16_T)
+            for(int i = 0; i < zoneNameLength; i++) {
+                zoneName[i] = (char16_t)ltime.tm_zone[i];
+            }
+        #elif defined(WCHAR_IS_WCHAR_T)
+            mbstowcs( (wchar_t*) zoneName, ltime.tm_zone, 
+                      sizeof(zoneName) / sizeof(WCHAR));
+        #else
+        #error "WCHAR should be either wchar_t or char16_t"
+        #endif
+
+            zoneName[zoneNameLength] = (WCHAR)0;
+            lastTimeZoneUpdateTickCount = tickCount;
+        }
+    }
+
+    // DateTimeHelper ******
+    #define TM_TO_SYSTIME(SYS, TM)           \
+            SYS->wYear = TM->tm_year + 1900; \
+            SYS->wMonth = TM->tm_mon + 1;  \
+            SYS->wDayOfWeek = TM->tm_wday;   \
+            SYS->wDay = TM->tm_mday;         \
+            SYS->wHour = TM->tm_hour;        \
+            SYS->wMinute = TM->tm_hour;      \
+            SYS->wSecond = TM->tm_hour
+
+    #define SYSTIME_TO_TM(SYS, TM)              \
+            TM->tm_year = SYS->wYear - 1900; \
+            TM->tm_mon = SYS->wMonth - 1;  \
+            TM->tm_wday = SYS->wDayOfWeek;   \
+            TM->tm_mday = SYS->wDay;         \
+            TM->tm_hour = SYS->wHour;        \
+            TM->tm_min = SYS->wMinute;       \
+            TM->tm_sec = SYS->wSecond
+
+    const bool SysLocalToUtc(SYSTEMTIME *local, SYSTEMTIME *utc)
+    {
+        struct tm local_tm;
+        SYSTIME_TO_TM(local, (&local_tm));
+
+        // tm doesn't have milliseconds
+        // uint32_t msec = utc->wMilliseconds;
+        // simply keep it as is
+
+        time_t utime = mktime(&local_tm);
+        
+        struct tm *utc_tm;
+        utc_tm = gmtime (&utime);
+
+        TM_TO_SYSTIME(utc, utc_tm);
+
+        // put milliseconds back
+        utc->wMilliseconds = local->wMilliseconds;
+
+        return true;
+    }
+
+    const bool SysUtcToLocal(SYSTEMTIME *utc, SYSTEMTIME *local, int64_t &bias, int &offset, bool &isDaylightSavings)
+    {
+        struct tm utc_tm;
+        SYSTIME_TO_TM(utc, (&utc_tm));
+
+        // tm doesn't have milliseconds
+        // uint32_t msec = utc->wMilliseconds;
+        // simply keep it as is
+
+        time_t ltime = mktime(&utc_tm);
+        
+        struct tm local_tm;
+        
+        tzset();
+        localtime_r(&ltime, &local_tm);
+        TM_TO_SYSTIME(local, (&local_tm));
+
+        // put milliseconds back
+        local->wMilliseconds = utc->wMilliseconds;
+
+        isDaylightSavings = local_tm.tm_isdst;
+
+        offset = local_tm.tm_gmtoff / 60;
+
+        struct tm *gtime = gmtime(&ltime);
+        time_t mtime = mktime(gtime);
+
+        bias = (int64_t)(mtime - ltime) / 60;
+
+        return true;
+    }
+
+    // DaylightTimeHelper ******
+    static void YmdToSystemTime(YMD* ymd, SYSTEMTIME *sys) noexcept
```

Comment:
This looks exactly similar to the win32 implementation- consider refactoring this to Platform/Common/DateTimeCommon.cpp